src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php

```diff
@@ -186,8 +187,12 @@ public function analyze(
         $project_analyzer = $this->getProjectAnalyzer();
 
         if ($codebase->track_unused_suppressions && !isset($storage->suppressed_issues[0])) {
-            foreach ($storage->suppressed_issues as $offset => $issue_name) {
-                IssueBuffer::addUnusedSuppression($this->getFilePath(), $offset, $issue_name);
+            if (count($storage->suppressed_issues) === 1
```

Comment:
Do you mind explaining what this condition does?

Reply:
That's there so `/** @psalm-suppress UnusedPsalmSuppress */` still shows up as an unused suppression. I could have done `array_values($storage->suppressed_issues) === ["UnusedPsalmSuppress"]` but I figured this is both faster and allows other cases to early out as well. Should I add a comment there so it's a bit more obvious?

Reply:
Oh! Yes please, it wasn't immediately obvious to me and I had the context :p

But yeah, I see what this does now, thanks!